spec/lib/diaspora/federated/contact_retraction_spec.rb

```diff
@@ -0,0 +1,63 @@
+describe ContactRetraction do
+  let(:contact) { FactoryGirl.build(:contact) }
+  let(:retraction) { ContactRetraction.for(contact) }
+
+  describe "#subscribers" do
+    it "contains the contact person" do
+      expect(retraction.subscribers).to eq([contact.person])
+    end
+  end
+
+  describe "#data" do
+    it "contains the hash with all data from the federation-retraction" do
+      federation_retraction_data = Diaspora::Federation::Entities.contact(contact).to_h
+
+      expect(retraction.data).to eq(federation_retraction_data)
+    end
+  end
+
+  describe ".retraction_data_for" do
+    it "creates a retraction for a contact" do
+      expect(Diaspora::Federation::Entities).to receive(:contact).with(contact).and_call_original
+      data = ContactRetraction.retraction_data_for(contact)
+
+      expect(data[:author]).to eq(contact.user.diaspora_handle)
+      expect(data[:recipient]).to eq(contact.person.diaspora_handle)
+      expect(data[:sharing]).to be_falsey
+      expect(data[:following]).to be_falsey
+    end
+  end
+
+  describe ".for" do
+    it "creates a retraction for a contact" do
+      expect(ContactRetraction).to receive(:retraction_data_for).with(contact)
+      ContactRetraction.for(contact)
+    end
+
+    it "sets receiving to false" do
+      expect(contact).to receive(:receiving=).ordered.with(false)
```

Comment:
Maybe it's better to check resulting retraction data value instead of checking for the function call?

The important thing here is that we don't create contact data with sharing/following with `true`, it's not that important how exactly we achieve this.

Also our `contact` object here has `sharing` and `receiving` values with `false` which is not a real world case, so this should probably be changed at least for this example or even maybe for the whole `ContactRetraction` test.